oap-server/server-cluster-plugin/cluster-etcd-plugin/src/main/java/org/apache/skywalking/oap/server/cluster/plugin/etcd/EtcdCoordinator.java

```diff
@@ -73,15 +83,30 @@ public EtcdCoordinator(ClusterModuleEtcdConfig config, EtcdClient client) {
                     if (!address.equals(selfAddress)) {
                         address.setSelf(false);
                     }
-                    res.add(new RemoteInstance(address));
+                    remoteInstances.add(new RemoteInstance(address));
                 });
             }
-
-        } catch (Exception e) {
+            if (remoteInstances.size() > 1) {
+                Set<String> remoteAddressSet = remoteInstances.stream().map(remoteInstance ->
+                        remoteInstance.getAddress().getHost()).collect(Collectors.toSet());
+                boolean hasUnHealthAddress = OAPNodeChecker.hasUnHealthAddress(remoteAddressSet);
+                if (hasUnHealthAddress) {
+                    this.healthChecker.unHealth(new ServiceQueryException("found 127.0.0.1 or localhost in cluster mode"));
+                } else {
+                    List<RemoteInstance> selfInstances = remoteInstances.stream().
+                            filter(remoteInstance -> remoteInstance.getAddress().isSelf()).collect(Collectors.toList());
+                    if (CollectionUtils.isNotEmpty(selfInstances) && selfInstances.size() == 1) {
```

Comment:
Why `&& selfInstances.size() == 1`? Are you planning to check duplicate addresses? If so, please move this login into `OAPNodeChecker`, all coordinator implementations should check this too, right?

Reply:
Yes, changed